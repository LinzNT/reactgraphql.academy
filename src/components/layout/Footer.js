import React from 'react'
import styled from 'styled-components'

import ContactForm from '../form/Contact'
import Grid, { Col, Row } from './Grid'
import ReactJSAcademyLogo from '../logos/ReactJSAcademy'
import Ul, { Li } from './Ul'
import DefaultLink from '../navigation/Link'
import { BLUE2, WHITE, BROWN } from '../../styles'

const StyledFooter = styled.footer`
  background-color: ${BLUE2};
  padding: 60px 0 40px 0;
  color: ${WHITE};
  h2 {
    color: ${WHITE};
  }
`

const Link = styled(DefaultLink)`
  color: ${WHITE};
`

const LinkList = styled(Ul)`
  padding-left: 0;
  list-style: none;
  margin-left: 0;
`

const Footer = () => (
  <Grid>
    <StyledFooter>
      <Row>
        <Col lg={5} lgOffset={1}>
          <ContactForm />
        </Col>
        <Col lg={4} lgOffset={1}>
          <Row>
            <Col md={6}>
              <LinkList>
                <Li>
                  <Link>Corporate Training</Link>
                </Li>
                <Li>
                  <Link>Sponsors & Partners</Link>
                </Li>
              </LinkList>
            </Col>
            <Col md={6}>
              <LinkList>
                <Li>
                  <Link>Privacy Policy</Link>
                </Li>
              </LinkList>
            </Col>
          </Row>
          <Row>
            <Col md={6}>
              Copyright &copy; 2018, ReactJS Academy is a{' '}
              <Link to="https://leanjs.com">LeanJS</Link> product
            </Col>
            <Col md={6}>
              <ReactJSAcademyLogo />
            </Col>
          </Row>
        </Col>
      </Row>
    </StyledFooter>
  </Grid>
)

export default Footer
