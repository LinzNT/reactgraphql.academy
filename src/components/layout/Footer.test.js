import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Footer from './Footer'

describe('<Footer />', () => {
  it('should render component', () => {
    const wrapper = shallow(<Footer />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
