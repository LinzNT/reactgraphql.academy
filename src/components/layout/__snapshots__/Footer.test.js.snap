// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render component 1`] = `
<styled.div>
  <Grid>
    <footer>
      <Row>
        <Styled(Col)
          md={5}
          mdOffset={1}
        >
          <withRouter(ContactForm)
            addContactUsLink={true}
          />
        </Styled(Col)>
        <Styled(Col)
          md={4}
          mdOffset={1}
        >
          <Row>
            <Styled(Col)
              md={6}
            >
              <Styled(styled.ul)>
                <styled.li>
                  <Link
                    to="/curriculum"
                  >
                    Curriculum
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    to="/react-redux-graphql-bootcamp"
                  >
                    React bootcamp
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    to="/about-us#corporate-team-training"
                  >
                    Corporate Training
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    to="/react-native-bootcamp"
                  >
                    React Native
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    to="/react-redux-graphql-part-time-course"
                  >
                    Part-time course
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    to="/blog"
                  >
                    Blog
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    to="/react-redux-graphql-training"
                  >
                    Training list
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    style={
                      Object {
                        "fontSize": "16px",
                      }
                    }
                    to="mailto:hello@reactjs.academy"
                  >
                    hello@reactjs.academy
                  </Link>
                </styled.li>
              </Styled(styled.ul)>
            </Styled(Col)>
            <Styled(Col)
              md={6}
            >
              <Styled(styled.ul)>
                <styled.li>
                  <Link
                    to="/about-us"
                  >
                    About us
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    to="/community"
                  >
                    Community
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    to="/privacy-policy"
                  >
                    Privacy Policy
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    to="/terms-of-service"
                  >
                    Terms of service
                  </Link>
                </styled.li>
                <styled.li>
                  <Link
                    to="/code-of-conduct"
                  >
                    Code of conduct
                  </Link>
                </styled.li>
              </Styled(styled.ul)>
              <styled.p>
                Follow us...
              </styled.p>
              <Styled(styled.ul)
                inline={true}
                unstyled={true}
              >
                <styled.li>
                  <Styled(Link)
                    to="https://twitter.com/reactjsacademy"
                  >
                    <BulletIcon
                      icon={[Function]}
                      social={true}
                    />
                  </Styled(Link)>
                </styled.li>
                <styled.li>
                  <Styled(Link)
                    to="https://www.instagram.com/reactjsacademy/"
                  >
                    <BulletIcon
                      icon={[Function]}
                      social={true}
                    />
                  </Styled(Link)>
                </styled.li>
                <styled.li>
                  <Styled(Link)
                    to="https://www.facebook.com/reactjsacademy/"
                  >
                    <BulletIcon
                      icon={[Function]}
                      social={true}
                    />
                  </Styled(Link)>
                </styled.li>
              </Styled(styled.ul)>
            </Styled(Col)>
          </Row>
          <Row>
            <Styled(Col)
              md={6}
            >
              <styled.p
                sm={true}
              >
                Copyright Â© 
                2017 - 2019
                , ReactJS Academy is a
                 
                <Link
                  to="https://leanjs.com"
                >
                  LeanJS
                </Link>
                 product
              </styled.p>
            </Styled(Col)>
            <Styled(Col)
              md={6}
            >
              <RJSALogo />
            </Styled(Col)>
          </Row>
        </Styled(Col)>
      </Row>
    </footer>
  </Grid>
</styled.div>
`;
